common/src/main/java/com/wynntils/mc/utils/objects/ServerIcon.java

```diff
@@ -112,8 +116,8 @@ private synchronized void loadServerIcon(ResourceLocation destination) {
         Validate.validState(nativeImage.getHeight() == 64, "Must be 64 pixels high");
 
         synchronized (this) {
-            serverIconLocation = destination;
             Minecraft.getInstance().getTextureManager().register(destination, new DynamicTexture(nativeImage));
+            serverIconLocation = destination;
```

Comment:
Why does this fix the race condition?

Reply:
Ah, I didn't fully read the code that renders the serverIcon, thought it was about setting the icon location before registering it. Anyways, even if this does not improve anything, it certainly won't cause harm.

Reply:
Perhaps, the DynamicTexture constructor does some rendering task that is the culprit.

Reply:
But it is also weird, the crash log in PR description hints that `nativeImage` is null when DynamicTexture is constructed. How can that even happen?

Reply:
The crash log doesn't mention `ServerIcon` at all.